### PR TITLE
Add initial migrations for db version 4

### LINKED ISSUE #701: Migrations still seem not to work
```
java.lang.IllegalStateException: Migration didn't properly handle Kino(de.tum.in.tumcampusapp.models.tumcabe.Kino).
Expected:
TableInfo{name='Kino', columns={rating=Column{name='rating', type='TEXT', notNull=true, primaryKeyPosition=0}, link=Column{name='link', type='TEXT', notNull=true, primaryKeyPosition=0}, trailer=Column{name='trailer', type='TEXT', notNull=false, primaryKeyPosition=0}, date=Column{name='date', type='TEXT', notNull=true, primaryKeyPosition=0}, actors=Column{name='actors', type='TEXT', notNull=true, primaryKeyPosition=0}, year=Column{name='year', type='TEXT', notNull=true, primaryKeyPosition=0}, cover=Column{name='cover', type='TEXT', notNull=true, primaryKeyPosition=0}, runtime=Column{name='runtime', type='TEXT', notNull=true, primaryKeyPosition=0}, genre=Column{name='genre', type='TEXT', notNull=true, primaryKeyPosition=0}, title=Column{name='title', type='TEXT', notNull=true, primaryKeyPosition=0}, description=Column{name='description', type='TEXT', notNull=true, primaryKeyPosition=0}, id=Column{name='id', type='TEXT', notNull=true, primaryKeyPosition=1}, director=Column{name='director', type='TEXT', notNull=true, primaryKeyPosition=0}, created=Column{name='created', type='TEXT', notNull=true, primaryKeyPosition=0}}, foreignKeys=[], indices=[]}
Found:
TableInfo{name='Kino', columns={rating=Column{name='rating', type='VARCHAR', notNull=false, primaryKeyPosition=0}, link=Column{name='link', type='TEXT', notNull=false, primaryKeyPosition=0}, trailer=Column{name='trailer', type='TEXT', notNull=false, primaryKeyPosition=0}, date=Column{name='date', type='VARCHAR', notNull=false, primaryKeyPosition=0}, actors=Column{name='actors', type='TEXT', notNull=false, primaryKeyPosition=0}, year=Column{name='year', type='VARCHAR', notNull=false, primaryKeyPosition=0}, cover=Column{name='cover', type='TEXT', notNull=false, primaryKeyPosition=0}, runtime=Column{name='runtime', type='VARCHAR', notNull=false, primaryKeyPosition=0}, genre=Column{name='genre', type='VARCHAR', notNull=false, primaryKeyPosition=0}, title=Column{name='title', type='TEXT', notNull=false, primaryKeyPosition=0}, description=Column{name='description', type='TEXT', notNull=false, primaryKeyPosition=0}, id=Column{name='id', type='INTEGER', notNull=false, primaryKeyPosition=1}, director=Column{name='director', type='TEXT', notNull=false, primaryKeyPosition=0}, created=Column{name='created', type='VARCHAR', notNull=false, primaryKeyPosition=0}}, foreignKeys=[], indices=[]}
at de.tum.in.tumcampusapp.database.TcaDb_Impl$1.validateMigration(TcaDb_Impl.java:309)
at android.arch.persistence.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:75)
at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:118)
at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:256)
at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:163)
at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:93)
at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:54)
at android.arch.persistence.room.RoomDatabase.compileStatement(RoomDatabase.java:204)
at android.arch.persistence.room.SharedSQLiteStatement.createNewStatement(SharedSQLiteStatement.java:65)
at android.arch.persistence.room.SharedSQLiteStatement.getStmt(SharedSQLiteStatement.java:72)
at android.arch.persistence.room.SharedSQLiteStatement.acquire(SharedSQLiteStatement.java:87)
at de.tum.in.tumcampusapp.database.dao.CafeteriaDao_Impl.removeCache(CafeteriaDao_Impl.java:76)
at de.tum.in.tumcampusapp.database.TcaDb.resetDb(TcaDb.java:186)
at de.tum.in.tumcampusapp.activities.StartupActivity.setupNewVersion(StartupActivity.java:308)
at de.tum.in.tumcampusapp.activities.StartupActivity.init(StartupActivity.java:99)
at de.tum.in.tumcampusapp.activities.StartupActivity.bridge$lambda$0$StartupActivity(StartupActivity.java)
at de.tum.in.tumcampusapp.activities.StartupActivity$$Lambda$2.run(Unknown Source)
at java.lang.Thread.run(Thread.java:760)
```

I'm really not sure, why room is such a headache

### PR DESCRIPTION
**Pull request by anselm92**
For some reason when updating from 1.5.1 	to 1.5.2 and then 1.5.3
the app crashed because the db used the initial preroom schema again.
Add migrations for current db version that make sure db uses correct schema.

Fixes #701

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/database/migrations/Migration3to4.kt [hunk 1]
```diff
@@ -10,6 +10,7 @@ import android.arch.persistence.room.migration.Migration
  */
 class Migration3to4 : Migration(3, 4) {
     override fun migrate(database: SupportSQLiteDatabase) {
+        Migration2to3().migrate(database)
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/database/migrations/Migration3to4.kt]: Can we really do this unconditionally? What happens on clean installing 1.5.2, then updating to 1.5.3 with that mirgation?

**Comment by anselm92** [comment]
@kordianbruck Geez sorry, forgot to do it, sure
@pfent i'll test that.

**Comment by pfent** [comment]
Probably a proper unit test for the whole migration path would be awesome:
https://medium.com/google-developers/testing-room-migrations-be93cdb0d975

I'm thinking about instantiating a database with the last non-room schema, then applying the migrations in order... @kordianbruck makes sense?

**Comment by kordianbruck** [comment]
@pfent perfect

**Comment by pfent** [comment]
Any idea how to extract that schema?

**Comment by kordianbruck** [comment]
@pfent install 1.5.1, adb pull with root rights the sqlite db from the data folder and export that as sql

**Pull request closed without merging**